### DISCUSSION TITLE
[New react template] No space left on device, write

### DISCUSSION THREAD

**Discussion by FlorentD** [beginner-corp/begin-community#104] [Support] [answered]
Hello Begin team !  👋

I'm struggling with my build process on my projects. I've got a `No space left on device, write` error on CI.
I've tried to setup a brand new project with your React template but it seems I've got the same :-( https://github.com/FlorentD/build-test

I've another private project on Begin with same error, but when I check `node_modules` size it's about 350Mo. Is the limit still 500Mo ?

Is disk space shared between project ? 

Is my only one solution to take off more dependences of my project ? :-)

Thank you for your work ! 

Florent

**Comment by tbeseda** [comment]
Hi Florent,
Thanks for the report and steps to reproduce. I will take a look today and update here

**Reply by tbeseda** [reply]
I was able to deploy the Node Create React App sample app with Begin
https://github.com/tbeseda/test-node-create-react-app deployed --> https://doge-laz.begin.app/
It looks the same as the one you linked 🤔 

As for disk space: it's not shared between projects, so I'm not sure what's going on with your private project. The example above is ~413MB, so it's definitely cutting it close.

I will get some definite numbers on limits and get back to you.
@FlorentD can you retry the example repo you linked in Begin's dashboard?

**Reply by FlorentD** [reply]
Thank you @tbeseda, the build is back !
About my private project, this is its id `k6jzsykrddj6mw`. I cannot figure out where I'm wrong. I've tried to get the minimum of dependencies (basically `React`, `ReactDOM`, `ReactRouter`, `Formik`, `Lodash`, `ol`, and `MaterialUI`) but I've got the same error on `install` step : 
```
⚬ Hydrate Hydrating dependencies in 3 paths
⚬ Hydrate Hydrating project root/
⚬ Hydrate Hydrated project root/
× Error: Command failed: npm ci 
npm WARN tar ENOSPC: no space left on device, write
``` 
It used to work with all this dependences on Begin.
I've tried to rollback `react-scripts` from `v4` to `v3` without success.

You've said the the default React repo takes already `~413MB` ? It means there is pretty much no space for any other libs ?

**Reply by tbeseda** [reply]
I did some more research into Begin deploy limits and the reasoning:
There's a cap on disk space because we use a Lambda to build, test, and deploy your app from the Lambda's /tmp directory. That directory has a hard cap of 512mb. We realize this is in conflict with some common stacks today. And we're actively working to solve this.

Anything in your project's package.json dependencies and devDependencies will be installed while building and deploying.

We have explored techniques that involve a sibling repo that builds into a main Begin/Architect project. Or even 2 projects, one for front end and one for the API server; but this may not work for your project where the bulk of dependencies are React related. 
If you're interested, I can loop in some teammates with more experience here. (Also, feel free to email support@begin.com if you'd like to avoid posting details in GH Discussions.)

Sorry I don't have better news. We are aware that this causes a problem for all kinds of projects and we really want to solve it ASAP for Begin users.

**Comment by FlorentD** [comment, marked as answer]
Thank you @tbeseda for your response.
Indeed it might be a problem for many projects starting and growthing into the Begin ecosystem.

I feel sorry because I like Begin but I'll have to move my app on another provider (Vercel). I need to update the app pretty often.

Hope you will solve it very soon !
